tests/Microsoft.Identity.Test.E2e/KeyGuardAttestationTests.cs

```diff
@@ -0,0 +1,169 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.Identity.Client.MtlsPop.Attestation;
+using Microsoft.Identity.Test.Common.Core.Helpers;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+using System;
+using System.Runtime.InteropServices;
+using System.Security.Cryptography;
+using Microsoft.Identity.Client.MtlsPop;
+using System.Threading.Tasks;
+using System.Threading;
+
+namespace Microsoft.Identity.Test.E2E
+{
+    [TestClass]
+    public class KeyGuardAttestationTests
+    {
+        private static CngKey CreateKeyGuardKey(string keyName)
+        {
+            const string ProviderName = "Microsoft Software Key Storage Provider";
+            const int NCRYPT_USE_VIRTUAL_ISOLATION_FLAG = 0x00020000;
+            const int NCRYPT_USE_PER_BOOT_KEY_FLAG = 0x00040000;
+
+            var p = new CngKeyCreationParameters
+            {
+                Provider = new CngProvider(ProviderName),
+                ExportPolicy = CngExportPolicies.None,
+                KeyUsage = CngKeyUsages.AllUsages,
+                KeyCreationOptions =
+                    CngKeyCreationOptions.OverwriteExistingKey |
+                    (CngKeyCreationOptions)NCRYPT_USE_VIRTUAL_ISOLATION_FLAG |
+                    (CngKeyCreationOptions)NCRYPT_USE_PER_BOOT_KEY_FLAG,
+            };
+
+            // Set 2048-bit RSA length
+            p.Parameters.Add(new CngProperty(
+                "Length",
+                BitConverter.GetBytes(2048),
+                CngPropertyOptions.None));
+
+            return CngKey.Create(CngAlgorithm.Rsa, keyName, p);
+        }
+
+        private static bool IsKeyGuardProtected(CngKey key)
+        {
+            try
+            {
+                // KeyGuard exposes a "Virtual Iso" property that is non-zero when protected.
+                // Same check used in #5448. :contentReference[oaicite:1]{index=1}
+                var prop = key.GetProperty("Virtual Iso", CngPropertyOptions.None);
+                var bytes = prop.GetValue();
+                return bytes != null && bytes.Length >= 4 && BitConverter.ToInt32(bytes, 0) != 0;
+            }
+            catch
+            {
+                return false;
+            }
+        }
+
+        [TestCategory("MI_E2E_AzureArc")]
```

Comment:
nit: pls add a comment on why we only run these tests on an specific machine  / Arc